src/Features/Core/Portable/QualifyMemberAccess/AbstractQualifyMemberAccessDiagnosticAnalyzer.cs

```diff
@@ -152,7 +152,8 @@ private bool IsStaticMemberOrTargetMethod(IOperation operation)
             switch (operation)
             {
                 case IMemberReferenceOperation memberReferenceOperation:
-                    return memberReferenceOperation.Member == null || memberReferenceOperation.Member.IsStatic;
+                    return memberReferenceOperation.Member == null || memberReferenceOperation.Member.IsStatic ||
+                        memberReferenceOperation.Member is IMethodSymbol method && method.MethodKind == MethodKind.LocalFunction;
                 case IInvocationOperation invocationOperation:
                     return invocationOperation.TargetMethod == null || invocationOperation.TargetMethod.IsStatic;
```

Comment:
Do we also need to handle the same case for the second switch section, i.e.

```
using System;
class C
{
    void Method()
    {
        object LocalFunction() => new object();
        |LocalFunction|]();
    }
    void Method2(Func<object> LocalFunction)
    {
    }
}
```

Reply:
Fixed!